### PR TITLE
feat: add user-scoped bookmark schema and server validation baseline

### PR DESCRIPTION
**Pull request by trystan2k**
## Summary

Establishes the data layer, server-side validation, and auth-scoping foundation for the bookmarks feature. Adds user-scoped bookmark/label Drizzle schema with cascade foreign keys, a typed validation pipeline for bookmark and label payloads, an authenticated TanStack Start middleware enforcing per-user data access, and a unified DB bootstrap script covering all migrations.

### Database schema (`src/db/schema/`)
- `bookmarks.ts` — `bookmark`, `label`, and `bookmark_label` tables scoped to `user.id` via cascade foreign keys; indexes on `user_id`, unique composite indexes (`user_id`+`name`, `bookmark_id`+`label_id`), state CHECK constraint, and full Drizzle relations.
- `bookmark-state.ts` — `active`/`archived` enum, default state, and type guard.
- `schema.ts` — barrel aggregator re-exporting auth + bookmark schema; wired into `drizzle.config.ts`, `src/db/client.ts`, and `auth.server.ts`.
- `drizzle/0002_violet_union_jack.sql` + `0002_snapshot.json` — generated migration for the new tables.

### Server-side validation (`src/features/bookmarks/lib/`)
- `validation.ts` — reusable primitives: discriminated `ValidationResult` union, field-error accumulator, `invalidValue` sentinel, and normalizers for required/optional strings, booleans, HTTP(S) URLs, and dates (with calendar-prefix sanity check).
- `bookmark-validation.ts` — `validateCreateBookmarkInput` / `validateUpdateBookmarkInput` with per-field length limits and state/favorite normalization.
- `label-validation.ts` — `validateCreateLabelInput` / `validateUpdateLabelInput` with name/color length limits.

### Auth scoping (`src/features/auth/server/`)
- `authenticated-middleware.ts` — TanStack Start `createMiddleware` that resolves the session and injects `{ session, userId }` into context; `createAuthenticatedServerFn` factory; returns 401 JSON on missing session.

### User-scoped data access (`src/features/bookmarks/server/`)
- `user-scope.server.ts` — `getBookmarkByIdForUser`, `requireBookmarkByIdForUser`, `getLabelByIdForUser`, and `assertBookmarkLabelOwnership`; all queries filter by `userId` to prevent cross-user access; throws `UserScopeAccessError` on miss.

### Bootstrap consolidation
- `scripts/bootstrap-db.mjs` — canonical migration runner (replaces `bootstrap-auth-db.mjs`, now a thin alias).
- Updated `package.json` scripts (`predev`, `pretest`, `preview:e2e`, CI workflow) to `pnpm db:bootstrap`; `db:bootstrap:auth` kept as compatibility alias.

## Tests

- `test/db/bookmark-schema.test.ts` — schema migration/constraint verification.
- `test/features/bookmarks/lib/bookmark-validation.test.ts` — create/update validation edge cases.
- `test/features/bookmarks/lib/label-validation.test.ts` — label validation edge cases.
- `test/features/bookmarks/server/user-scope.server.test.ts` — user-scoped lookup + ownership enforcement.
- `test/features/auth/server/authenticated-middleware.test.ts` — middleware session resolution and 401 handling.
- `test/lib/bootstrapped-temp-db.ts` — shared helper for fresh bootstrapped temp DBs (refactored out of `auth-bootstrap.test.ts`).

## Test results

- ✅ Pre-commit: oxlint + oxfmt (lint-staged) passed
- ✅ Pre-commit: `tsc --noEmit` passed
- ✅ Pre-push: production build passed

### CODE REVIEW

File: src/db/client.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { createClient } from '@libsql/client';
 import { drizzle } from 'drizzle-orm/libsql';
 
 import { resolveDatabaseCredentials } from '@/db/database-url';
-import * as schema from '@/db/schema/auth';
+import * as schema from '@/db/schema/schema';
 import { getAuthEnvironment } from '@/features/auth/server/env.server';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply src/db/client.ts:6]: Revalidated this against the current `@libsql/client` behavior. The same `createClient(resolveDatabaseCredentials(...))` path used in `src/db/client.ts` already starts with `PRAGMA foreign_keys = 1`, and FK / `ON DELETE CASCADE` behavior is enforced without an extra pragma call. Because the runtime connection already has foreign-key enforcement enabled, no code change is needed here.

File: scripts/bootstrap-db.mjs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  const client = createClient(databaseCredentials);
+
+  try {
+    const db = drizzle(client);
+
+    await migrate(db, {
+      migrationsFolder: path.join(repoRoot, 'drizzle')
+    });
+  } finally {
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply scripts/bootstrap-db.mjs:35]: Revalidated the bootstrap path as well. The `createClient(databaseCredentials)` connection used in `scripts/bootstrap-db.mjs` also comes up with `PRAGMA foreign_keys = 1`, so migrations already run with FK enforcement active. Given that the bootstrap connection already has the expected semantics, no additional code change is needed in this script.

File: test/features/bookmarks/server/user-scope.server.test.ts [hunk 3]
```diff
@@ -0,0 +1,339 @@
+import { afterEach, beforeEach, describe, expect, test } from 'vitest';
+
+import { drizzle } from 'drizzle-orm/libsql';
+
+import type { createDb } from '@/db/client';
+import * as schema from '@/db/schema/schema';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply test/features/bookmarks/server/user-scope.server.test.ts:6]: Revalidated this against the current repo config and current file state. With the current `tsconfig.json` and `verbatimModuleSyntax` settings, `pnpm typecheck` passes and the pattern in this test file does not reproduce a TypeScript error here. Since the reported compiler break does not occur on the current codebase, we're not applying a code change for this thread.

File: test/features/bookmarks/server/user-scope.server.test.ts [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+} from '@/features/bookmarks/server/user-scope.server';
+import {
+  createBootstrappedTempDatabase,
+  disposeBootstrappedTempDatabase,
+  type BootstrappedTempDatabase
+} from '../../../lib/bootstrapped-temp-db';
+
+type TempDatabaseContext = BootstrappedTempDatabase & {
+  db: ReturnType<typeof createDb>;
+};
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply test/features/bookmarks/server/user-scope.server.test.ts:31]: Revalidated this against the current repo config and current file state. With the current `tsconfig.json` and `verbatimModuleSyntax` settings, `pnpm typecheck` passes and the pattern in this test file does not reproduce a TypeScript error here. Since the reported compiler break does not occur on the current codebase, we're not applying a code change for this thread.

File: e2e/tests/smoke.spec.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import { expect, test } from '../fixtures/test';
-import { appRoutes } from '../utils/routes';
+import { expect, test } from '@e2e/fixtures/test';
+import { appRoutes } from '@e2e/utils/routes';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply e2e/tests/smoke.spec.ts:2]: No, it works.

File: e2e/tests/not-found-smoke.spec.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
-import { expect, test } from '../fixtures/test';
-import { appRoutes } from '../utils/routes';
+import { expect, test } from '@e2e/fixtures/test';
+import { appRoutes } from '@e2e/utils/routes';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply e2e/tests/not-found-smoke.spec.ts:2]: No, it works.

File: e2e/tests/auth-error-smoke.spec.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
-import { expect, test } from '../fixtures/test';
-import { appRoutes } from '../utils/routes';
+import { expect, test } from '@e2e/fixtures/test';
+import { appRoutes } from '@e2e/utils/routes';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply e2e/tests/auth-error-smoke.spec.ts:2]: No, it works.

File: e2e/tests/auth/authenticated-smoke.spec.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
-import { expect, test } from '../../fixtures/test';
-import { appRoutes } from '../../utils/routes';
+import { expect, test } from '@e2e/fixtures/test';
+import { appRoutes } from '@e2e/utils/routes';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply e2e/tests/auth/authenticated-smoke.spec.ts:2]: No, it works.

File: e2e/tests/auth/authenticated-not-found-smoke.spec.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
-import { expect, test } from '../../fixtures/test';
-import { appRoutes } from '../../utils/routes';
+import { expect, test } from '@e2e/fixtures/test';
+import { appRoutes } from '@e2e/utils/routes';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply e2e/tests/auth/authenticated-not-found-smoke.spec.ts:2]: No, it works.

File: e2e/tests/auth/google-oauth.spec.ts [hunk 10]
```diff
@@ -1,5 +1,5 @@
-import { expect, test } from '../../fixtures/test';
-import { appRoutes } from '../../utils/routes';
+import { expect, test } from '@e2e/fixtures/test';
+import { appRoutes } from '@e2e/utils/routes';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply e2e/tests/auth/google-oauth.spec.ts:2]: No, it works.

File: e2e/tests/locale-smoke.spec.ts [hunk 11]
```diff
@@ -1,5 +1,5 @@
-import { expect, test } from '../fixtures/test';
-import { appRoutes } from '../utils/routes';
+import { expect, test } from '@e2e/fixtures/test';
+import { appRoutes } from '@e2e/utils/routes';
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply e2e/tests/locale-smoke.spec.ts:2]: No, it works.

File: src/features/bookmarks/lib/validation.ts [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+export function createValidationFailure<TData, TField extends string>(
+  fieldErrors: ValidationFieldErrors<TField>,
+  formError?: string
+): ValidationResult<TData, TField> {
+  return {
+    fieldErrors,
+    formError,
+    success: false
+  };
+}
```
[the remark this answers is not in the document]
Comment (trystan2k) [reply src/features/bookmarks/lib/validation.ts:40]: It is just style, does not fix
